dtls.c

```diff
@@ -3923,8 +3928,7 @@ handle_0_verified_client_hello(dtls_context_t *ctx, dtls_ephemeral_peer_t *ephem
 
   err = handle_verified_client_hello(ctx, peer, data, data_length);
   if (err < 0) {
-    DEL_PEER(ctx->peers, peer);
-    dtls_free_peer(peer);
+    dtls_destroy_peer(ctx, peer, 0);
```

Comment:
At that point I guess, it's better to DTLS_DESTROY_CLOSE.
That indicates the other peer, that this peer doesn't continue (for whatever reason).

Reply:
Thanks, I was unsure about this one. I have changed this accordingly.